### ISSUE TITLE
Proposal: render the UI conditionally

### ISSUE THREAD

**Issue by marcofugaro** [gsimone/leva#20]
Sometimes, it is useful to show the pane conditionally, while everything works the same. In my case, I need to show the pane only to designers and not the client. I achieve this via a special url, and I have a check like 

```js
export const DEBUG = window.location.search.includes('?debug')
```

which check if `?debug` is appended to the url.

WIth `use-tweaks`, we achieved this with a really dirty `useTweaksOnDebug`:

```js
const getValuesFromTweaks = (...args: any[]) => {
  const obj = typeof args[0] === 'object' ? args[0] : args[1]

  const getEntries = (obj: any): any => {
    return Object.keys(obj).flatMap((k) => {
      const v = obj[k]

      if (v.type !== undefined) {
        if (v.type === 1) {
          // Folder
          return getEntries(v.schema)
        } else {
          // Button or Separator
          return []
        }
      } else {
        return [[k, v.value ?? v]] // array of tuple because of flatmap
      }
    })
  }

  return Object.fromEntries(getEntries(obj))
}

export const useTweaksOnDebug: typeof useTweaks = DEBUG ? useTweaks : getValuesFromTweaks
```

I don't know how the API of this would look like if we plan to remove the `<Twix>` component, any solution is appreciated.

**Comment by dbismut** [comment]
@marcofugaro how do you think we should work on this? simple fix would be to add `display: none` to the <LevaRoot /> component. I think that would be preferred as some new advanced features expose the store and would break if you would just return the initial data that way.

**Comment by marcofugaro** [comment]
It's fine to add `display: none` but it needs to be documented!

> I think that would be preferred as some new advanced features expose the store and would break if you would just return the initial data that way.

Exposing the store would be fine, the issue was just about not doing `ReactDOM.render()` /`document.body.appendChild()`. But now that we have the `<Leva />` component the situation is different.

**Comment by dbismut** [comment]
Oh sorry I thought you wanted to add built-in support for `?debug=1` in the url?

If this should remain in userland then I'll just add a ˋhidden` prop that does that for you.

**Comment by marcofugaro** [comment]
That would work perfectly!

`<Leva hidden={window.location.search.includes('debug')}`

🤗

**Comment by dbismut** [comment]
Great! Should be fixed by a7ce7fc, sorry for not making a PR!
